src/plugincode/post_scan.py

```diff
@@ -38,14 +38,30 @@
 
 
 @post_scan_spec
-def post_scan_handler(active_scans, results):
+class PostScanPlugin(object):
     """
-    Process the scanned files and yield the modified results.
-    Parameters:
-     - `active_scans`: a list of scanners names requested in the current run.
-     - `results`: an iterable of scan results for each file or directory.
+    A post-scan plugin layout class to be extended by the post_scan plugins.
     """
-    pass
+
+    def __init__(self, option, user_input):
+        self.option = option
+        self.user_input = user_input
+
+    def process_results(self, results, active_scans):
+        """
+        Process the scan results.
+        results - an iterable of resources
+        active_scans - iterable of scanners that were used to obtain the results (e.g. "copyrights", "licenses")
+        """
+        return results
+
+    @staticmethod
+    def get_click_options():
```

Comment:
calling this just `get_options` might be enough IMHO

Reply:
Agreed. Should I also change it on `PreScanPlugin`?

Reply:
Please do!

Reply:
eventually the class template should be the same for all plugins. And a plugin becomes a pre/scan/post/output plugin because it is declared to be so :) not because of its "shape"